test/src/test/java/hudson/util/ArgumentListBuilder2Test.java

```diff
@@ -61,4 +70,33 @@ public void slaveMask() throws Exception {
         System.out.println(out);
         assertTrue(out.toString().contains("$ java ********"));
     }
+
+    @Test
+    public void ensureArgumentsArePassedToCommandUnmodified() throws Exception {
+        String[] specials = new String[] {
+                "~", "!", "@", "#", "$", "%", "^", "&", "*", "(", ")",
+                "_", "+", "{", "}", "[", "]", ":", ";", "\"", "'", "\\", "|",
+                "<", ">", ",", ".", "/", "?", " "
+        };
+
+        ArgumentListBuilder args = new ArgumentListBuilder(JavaEnvUtils.getJreExecutable("java"), "-cp", "target/test-classes/", "hudson.util.EchoCommand");
```

Comment:
IMO this way is pretty enough to be merged. Probably it could make sense to move the check to a framework method in order to simplify the test.